std/container/rbtree.d

```diff
@@ -1806,6 +1849,41 @@ auto redBlackTree(alias less, bool allowDuplicates, E)(E[] elems...)
     return new RedBlackTree!(E, binaryFun!less, allowDuplicates)(elems);
 }
 
+
+import std.range.primitives: isInputRange, isImplicitlyConvertible, ElementType;
+
+/++ Ditto +/
+auto redBlackTree(Stuff)(Stuff range)
+if (isInputRange!Stuff && !is(typeof(range) == string))
```

Comment:
Instead of typeof(range) == string, you should use `isSomeString` to cover all string types.